Feature/build

aparnagurram1 opened the pull request:
## Summary
<!--- Describe in single paragraph product level requirements for this change -->
Automate Bless setup

## What kind of change does this PR introduce?
- Admin flow automation
   Upload Bless CA keys to s3 bucket
- Build and deploy bless CA
- Copy CA public key to snapcheck infrastructure
- User flow automation
   Shell script to invoke the bless client lambda
<!-- Please check the one that applies to this PR using "x" i.e. [x] -->

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behaviour?
<!-- Please describe in detail the current behaviour (bug) that you are fixing -->
SSH access setup is manual
## What is the new behaviour?
<!-- if this is a new feature provide links to project/product documentation -->
SSH access setup is automated

aparnagurram1 committed:
Automate bless-ca setup

Instead of packaging dependencies manually, we are automating this process

[Resolves: https://app.asana.com/0/1189821244930945/1189821244930958]

aparnagurram1 committed:
update node lable

aparnagurram1 committed:
Use python3.7

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Makefile

aparnagurram1 committed:
fix Makefile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
fix Makefile

aparnagurram1 committed:
fix Jenkinsfile

aparnagurram1 committed:
Fetch privatekey from s3

aparnagurram1 committed:
Automate the process of admin and user flow

Instead of manually implementing  bless CA lamnda, we are automating this process via jenkins pipeline
earlier inorder to call bless lambda we were using python bless client code to invoke the bless lambda and so in this case we have to provide the python client code to users. Inorder to simplify the user flow refactor the python client code to another bless_client lambda function so that now users can make api call to the client code which in turn makes api call to bless ca and gets signed certificate
[Resolves: https://app.asana.com/0/1189821244930945/1189821244930958]

mountainowl commented on Jenkinsfile:
```diff
@@ -0,0 +1,51 @@
+      steps {
+        sh 'cd bless'
+        sh 'aws s3 cp s3://blessca-keys/bless-ca- lambda_configs/bless-ca-'
+        //sh 'python3.7 -m venv venv'
+        //sh '. venv/bin/activate' 
```
remove commented out code; it creates confusion

mountainowl commented on ansible/ansible.cfg:
```diff
@@ -0,0 +1,7 @@
+[defaults]
+private_key_file=~/Downloads/snapcheck-non-prod-server.pem
```
not a great location; since it's based on your MAC 
change it to be universal for systems

mountainowl commented on ansible/inventories/dev/hosts:
```diff
@@ -0,0 +1,2 @@
+[linux]
```
at sometime we'll need to main inventory of our deployment in single file. 
not important rn

mountainowl commented on bless_client/fetch_cert.sh:
```diff
@@ -0,0 +1,28 @@
+
+echo "invoking aws bless client-lambda"
+
+aws lambda invoke --invocation-type RequestResponse --function-name bless_client --region us-east-1 --log-type Tail --payload file://./payload.json --cli-binary-format raw-in-base64-out id_rsa-cert.pub
```
I'm on with this rn.
We need to create API gateway on AWS, rather than invoking lambda directly. invoking lambda directly means AWS client is installed locally, which may not be the case.

mountainowl commented on lambda_configs/bless_deploy.cfg:
```diff
@@ -0,0 +1,65 @@
+[Bless CA]
+# You must set an encrypted private key password for each AWS Region you deploy into
+# for each aws region specify a config option like '{}_password'.format(aws_region)
+us-east-1_password = AQICAHibDsNZpMt+ZjdpBHrsbZd/fbjVqiF/CSncj9Y/Xvea3AH0LX6HgJQYQQsPtqkp96OvAAAAZjBkBgkqhkiG9w0BBwagVzBVAgEAMFAGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMfK7NV2NIPX7FzAtQAgEQgCNZPf6d5MKXW+tIon2DMIOtLpqtQHY+TCIGUwo6Dp0DOE+3Iw==
```
i believe there is a way around it i.e. putting this in env var or so; not important but just so that you know

aparnagurram1 committed:
Address code review commits

Make the public key path generic
[Resolves: https://app.asana.com/0/1189821244930945/1189821244930958]
